File: app/src/main/java/com/flysolo/dmmsugradelevelapp/utils/LoadingDialog.java
package com.flysolo.dmmsugradelevelapp.utils;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.TextView;

import androidx.appcompat.app.AlertDialog;

import com.flysolo.dmmsugradelevelapp.R;
import com.google.android.material.dialog.MaterialAlertDialogBuilder;

public class LoadingDialog {
    Context context;
    AlertDialog dialog;

    public LoadingDialog(Context context) {
        this.context = context;

    }
    public void  showLoadingDialog(String title) {
        MaterialAlertDialogBuilder builder = new MaterialAlertDialogBuilder(this.context);
        View view = LayoutInflater.from(context).inflate(com.flysolo.dmmsugradelevelapp.R.layout.loading_dialog,null);
        TextView titles = view.findViewById(R.id.textTitle);
        titles.setText(title);
        builder.setView(view);
        builder.setCancelable(false);
        dialog = builder.create();
        dialog.show();
    }
    public void  stopLoading() {
        dialog.dismiss();
    }
}

File: app/src/main/java/com/flysolo/myapplication/utils/LoadingDialog.kt
package com.flysolo.myapplication.utils

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.widget.TextView
import com.flysolo.myapplication.R

import com.google.android.material.dialog.MaterialAlertDialogBuilder

class LoadingDialog(private val context : Context) {
    var alertDialog: androidx.appcompat.app.AlertDialog ? = null
    fun showLoadingDialog(title : String){
        val materialAlertDialogBuilder = MaterialAlertDialogBuilder(context)
        val view : View = LayoutInflater.from(context).inflate(R.layout.loading_dialog,null)
        view.findViewById<TextView>(R.id.textTitle).text = title
        materialAlertDialogBuilder.setView(view)
        materialAlertDialogBuilder.setCancelable(false)
        alertDialog = materialAlertDialogBuilder.create()
        alertDialog?.show()
    }
    fun stopLoading(){
        alertDialog?.dismiss()
    }
}